cmd/operator/internal/controller/kubearchiveconfig_controller.go

```diff
@@ -254,7 +254,7 @@ func (r *KubeArchiveConfigReconciler) reconcileSinkRole(ctx context.Context, kac
 		resource := sourcesv1.APIVersionKindSelector{Kind: kar.Selector.Kind, APIVersion: kar.Selector.APIVersion}
 		resources = append(resources, resource)
 	}
-	return r.reconcileRole(ctx, kaconfig, k9eSinkName, r.getRules(ctx, resources, []string{"delete"}))
+	return r.reconcileRole(ctx, kaconfig, k9eSinkName, r.getRules(ctx, resources, []string{"list", "delete"}))
```

Comment:
It seems like `get` would be a better choice here than `list`.

Reply:
I disagree. Using `list` seems counter intuitive but it lets the sink distinguish between a situation where there is a permissions/networking issue and the object already is deleted. The sink does a list using a field selector, so if the object is deleted, we get an empty list. With `get`, if the object was deleted, the sink gets an error, which is same thing that happens if there is a permissions/networking issue

Reply:
That's what this check does:
```
if !errs.IsNotFound(getErr) {
```

Reply:
I'm confused what you're referring to. The code that I described is over [here](https://github.com/kubearchive/kubearchive/blob/651dcce1779e0f1c95e740a18e24471af91eecf9/cmd/sink/routers/routers.go#L213)

Reply:
Aren't you using the same dynamic client that is used in `test/integration/operator_test.go`?

Reply:
yes, I did not realize that `errs.IsNotFound(getErr)` was a utility function for check those kinds of errors. I will switch to using `get`

Reply:
actually, using `errs.IsNotFound(getErr)` I was able to make this work by checking the result of the already existing delete request. Please take a look